Sources/mtlswift/AST/CustomDeclaration.swift

```diff
@@ -34,8 +34,12 @@ public enum CustomDeclaration {
                 return nil
             }
 
-            let swiftTypeName = scanner.leftString
+            var swiftTypeName = scanner.leftString
                                        .trimmingCharacters(in: .whitespacesAndNewlines)
+            if #available(OSX 10.12, *),
+               let checkedType = TypeSanitizer.sanitizeType(swiftTypeName) {
```

Comment:
I think we can have it, at least as an experimental feature, but we defintely need to introduce a command line argument parameter for this

also, you have to keep in mind that users can use aliases/typenames of their private codebase and that is totally fine (we do not limit swift type usage to simd) hence your approach won't work. I think what we can do instead is introduce a simple dictionary here and try map `swiftTypeName` to something else